controllers/vaultdynamicsecret_controller.go

```diff
@@ -229,6 +229,13 @@ func (r *VaultDynamicSecretReconciler) Reconcile(ctx context.Context, req ctrl.R
 				"Secret synced, isStaticCreds=%t, horizon=%s, ttl=%d",
 				true, horizon, o.Status.StaticCredsMetaData.TTL)
 		}
+	} else {
+		// Not all dynamic credentials are renewable (e.g.: aws/creds/)
+		// In the scenario we resync the secret within the lease duration window
+		nextRenewalTime := nextRenewalTime(o)
+		horizon = nextRenewalTime.Sub(time.Now())
```

Comment:
It seems possible that `horizon` could be less than or equal to 0 seconds, in the case where `nextRenewalTime()` is in the past. We will want to handle that case to ensure that we keep syncing, as is done here https://github.com/hashicorp/vault-secrets-operator/pull/320/files/17552b646993eaf812add00d6759ac0a0febb786#diff-162ee5e0791fe53245a14c3a3eb3ddc1e296cad32b7081a91cecc0becdeb9b0dR223